Tests/UserServiceTests/Validators/UserEmailPasswordValidatorTests.cs

```diff
@@ -0,0 +1,420 @@
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.EmailEmpty);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissPassword()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                FirstName = "Leo",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.EmptyPassword);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissFirstName()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Password = "123",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.FirstNameEmpty);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void MissLastName()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Password = "123",
+                FirstName = "Leo",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.LastNameEmpty);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void EmailTooLong()
+        {
+            CreateUserRequest user = new CreateUserRequest
+            {
+                Email = new string('c', 50) + "@ya.ru",
+                Password = "123",
+                FirstName = "Leo",
+                LastName = "Kor",
+                Birthday = DateTime.Today
+            };
+
+            var validationResult = Assert.Throws<ValidationException>(() => validator.ValidateAndThrow(user));
+
+            var expectedError =
+                validationResult.Errors.FirstOrDefault(
+                    error => error.ErrorMessage == ErrorsMessages.EmailTooLong);
+
+            Assert.IsNotNull(expectedError);
+        }
+
+        [Test]
+        public void FirstNameTooLong()
```

Comment:
We can also add too short , doestn't start with big letter, containsforbidden symbols

Reply:
I have tests that cover these cases. There are FirstNameFirstLetterIsNotUpperCase, FirstNameContainsNumbers

Reply:
> I have tests that cover these cases. There are FirstNameFirstLetterIsNotUpperCase, FirstNameContainsNumbers

Not sure if it's perfect to do those checks in same test, but not critical

Reply:
Added test for too short, which called FirstName1Letter